packages/discovery-react-components/src/components/SearchFacets/components/FieldFacets.tsx

```diff
@@ -39,48 +40,46 @@ export const FieldFacets: FC<FieldFacetsProps> = ({
   collapsedFacetsCount,
   onChange
 }) => {
-  const handleOnChange = (
-    selectedFacetName: string,
-    selectedFacetKey: string,
-    checked: boolean
-  ): void => {
-    const facetsForNameIndex = allFacets.findIndex(facet => {
-      if (!facet.name) {
-        return facet.field === selectedFacetName;
-      }
-      return facet.name === selectedFacetName;
-    });
-    if (facetsForNameIndex > -1) {
-      const facetsForName = allFacets[facetsForNameIndex];
-      const multiselect = get(facetsForName, 'multiple_selections_allowed', true);
-      const facetResults: SelectableQueryTermAggregationResult[] = get(
-        facetsForName,
-        'results',
-        []
-      );
+  const handleOnChange = (selectedFacets: SelectedFacet[]): void => {
+    selectedFacets.map(selectedFacet => {
+      const facetsForNameIndex = allFacets.findIndex(facet => {
+        if (!facet.name) {
+          return facet.field === selectedFacet.selectedFacetName;
+        }
+        return facet.name === selectedFacet.selectedFacetName;
+      });
```

Comment:
Can you destruct? Something like - 

```suggestion
    selectedFacets.map(({ selectedFacetName, selectedFacetKey,  checked }) => {
      const facetsForNameIndex = allFacets.findIndex(({ field, name }) => {
        if (!name) {
          return field === selectedFacetName;
        }
        return name === selectedFacetName;
      });
```
selectedFacetKey,  checked will be used further down